### PR TITLE
Feature/slide design

### PR DESCRIPTION
**Pull request by kemaric**
This is the base implementation of the Slide and Test Session Model data. Feel free to suggest any changes or make comments.

The files to focus your review on are the following files in the directory **src/app/:**
Everything in js/slides
package.json
index.html
js/main.js

### CODE REVIEW

File: js/slidesModel.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+         */
+        this.options = [];
+        /**
+         * Breadcrum steps for the slide. These can be instructions for the user to follow.
+         */
+        this.steps = [];
+        /**
+         * [Optional] The answer to the question. 
+         */
+        this.answer = "";
```
Comment (kemaric) [inline js/slidesModel.js]: Change this to be `expectedAnswer` because this is confusing with `userReponse`.

File: js/slidesModel.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
+/**
+ * The base class for Slides.
+ */
+class BaseSlide {
+    constructor(label) {
+        if (!BaseSlide._id) { BaseSlide._id = 0; }
+        BaseSlide._id++;
+        this._uid = `${this.constructor.name}-${BaseSlide._id}`;
+        this._label = label;
```
Comment (kemaric) [inline js/slidesModel.js]: Label can be an object that has a category, and an optional ordering. ie {category: "Diffususion", order:1}.

**Review by kouissar — CHANGES_REQUESTED** [review]
Looks good to me. Please integrate the functionality so we could test and provide feedback based on the tested app.

**Review by kouissar — APPROVED** [review]

File: js/slides/slidesModel.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        super(label);
+        /**
+         * The question or text for the slide.
+         */
+        this.question = "";
+        /**
+         * [Optional] Options for the slide. If it's a test, then it would be possible options for the 
+         * question's answer.
+         */
+        this.options = [];
```
Comment (OliviaSims21) [inline js/slides/slidesModel.js:47]: This may not need to be necessary. However, keep for now incase it becomes useful when we start to insert tests.

**Review by kouissar — APPROVED** [review]
approved

**Review by kouissar — APPROVED** [review]

**Review by kouissar — APPROVED** [review]

**Review by kouissar — APPROVED** [review]
Approved

**Review by skatulich — APPROVED** [review]
Don't see any issues.

**Pull request merged**